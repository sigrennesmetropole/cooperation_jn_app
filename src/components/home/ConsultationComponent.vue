<script setup lang="ts">
import fabriqueCitoyenne from '@/assets/icons/fabrique-citoyenne.svg'
import router from '@/router'
import { usePanelsStore } from '@/stores/panels'
import { computed } from 'vue'

const panelStore = usePanelsStore()

const leftAlignment = computed(() =>
  panelStore.isInformationPanelShown
    ? 'absolute left-[462px] top-3'
    : 'absolute left-3 top-3'
)

function goToConsultation() {
  router.push('/consultation')
}
</script>

<template>
  <div
    class="z-10 flex flex-row gap-4 p-2 bg-[#261E69] rounded-lg text-white items-center"
    :class="leftAlignment"
  >
    <div class="flex flex-row gap-3 px-2">
      <img :src="fabriqueCitoyenne" class="w-9 h-9 my-auto" />
      <div>
        <h4 class="font-dm-sans font-bold text-lg leading-6">
          La Fabrique Citoyenne
        </h4>
        <p class="font-dm-sans font-normal text-sm">
          Découvrez l'espace concertation et <br />
          contribuez au transport de demain
        </p>
      </div>
    </div>
    <button
      tabindex="0"
      class="font-dm-sans font-medium text-base border border-white px-4 py-3 rounded-lg h-12"
      @click="goToConsultation()"
      @keydown.enter="goToConsultation()"
    >
      Découvrir
    </button>
  </div>
</template>
