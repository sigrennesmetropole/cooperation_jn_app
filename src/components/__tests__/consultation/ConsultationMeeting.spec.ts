import ConsultationComponent from '@/components/consultation/ConsultationMeeting.vue'
import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'

describe('ConsultationComponent', () => {
  const wrapper = mount(ConsultationComponent, {
    props: {
      illustration: 'src/assets/illustrations/ex_consultation.png',
      title: 'Vos réactions au futur aménagement du réseau trambus',
      status: 'open',
      date: '15/10/2023',
      place: 'Zone géographique',
      content:
        'Appaisement du traffic et végétalisation des espaces. La rue chicogné va faire l’objet d’une requalification',
      comment: 481,
      like: 1291,
      person: 353,
    },
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
