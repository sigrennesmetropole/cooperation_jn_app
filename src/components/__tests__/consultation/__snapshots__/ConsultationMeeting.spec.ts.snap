// Vitest Snapshot v1

exports[`ConsultationComponent > renders correctly 1`] = `
"<div class=\\"flex flex-row gap-4 py-4 border-b border-b-slate-200\\" illustration=\\"src/assets/illustrations/ex_consultation.png\\">
  <div class=\\"flex flex-col gap-2\\">
    <p class=\\"font-dm-sans font-bold text-base\\">Vos réactions au futur aménagement du réseau trambus</p>
    <div class=\\"flex flex-row gap-3 h-8\\"><button class=\\"bg-amber-400 flex items-center py-1 px-2 rounded h-[28px]\\">
        <div class=\\"font-medium font-dm-sans text-sm text-center items-center text-black\\">Ouverte</div>
      </button>
      <div class=\\"flex flex-row gap-1.5 items-center\\"><img src=\\"/src/assets/icons/calendar.svg\\" class=\\"w-[15px] h-[15px]\\">
        <p class=\\"font-dm-sans font-normal text-xs\\"> Cloturée le <br> 15/10/2023</p>
      </div>
      <div class=\\"flex flex-row gap-1.5 items-center\\"><img src=\\"/src/assets/icons/localization-icon.svg\\" class=\\"w-4 h-4\\">
        <p class=\\"font-dm-sans font-normal text-xs\\">Zone géographique</p>
      </div>
    </div>
    <p class=\\"font-dm-sans font-normal text-xs\\">Appaisement du traffic et végétalisation des espaces. La rue chicogné va faire l’objet d’une requalification</p>
    <div class=\\"flex flex-row gap-3 py-3\\">
      <div class=\\"flex flex-row gap-1.5 items-center\\"><img src=\\"/src/assets/icons/chat-bubble.svg\\" class=\\"w-4 h-4\\">
        <p>481</p>
      </div>
      <div class=\\"border border-l border-slate-200\\"></div>
      <div class=\\"flex flex-row gap-1.5 items-center\\"><img src=\\"/src/assets/icons/like-thumb.svg\\" class=\\"w-4 h-4\\">
        <p>1291</p>
      </div>
      <div class=\\"border border-l border-slate-200\\"></div>
      <div class=\\"flex flex-row gap-1.5 items-center\\"><img src=\\"/src/assets/icons/multiple-person.svg\\" class=\\"w-4 h-4\\">
        <p>353</p>
      </div>
    </div>
  </div>
</div>"
`;
